resource.address/spi/src/main/java/org/kaazing/gateway/resource/address/ResourceAddressFactorySpi.java

```diff
@@ -463,5 +471,36 @@ protected void parseNamedOptions0(URI location, ResourceOptions options,
     protected String getRootSchemeName() {
         return null;
     }
+    
+    /**
+     * Throw error on specific circumstances:
+     * - no addresses available for binding
+     * - when PreferedIPv4 flag is true and the host IP is IPV6
+     * @param location
+     */
+    private void throwNoAddressesToBindError(URI location) {
+        StringBuffer error = new StringBuffer(format(NO_ADDRESSES_AVAILABLE_FOR_BINDING, location));
+        try {
+            InetAddress address = InetAddress.getByName(location.getHost());
+            error.insert(0, getErrorMessageIfPreferedIPv4(address));
```

Comment:
personally I think it may be clearer to read to pull the method call getErrorMessageIfPreferedIPv4 into this block and use normal if block.  The return of `""` making this call a NOOP is unexpected.  Perhaps something like this would be better

```java
if("true".equals(System.getProperty(JAVA_NET_PREFER_IPV4_STACK)) && (address instanceof Inet6Address)){
    error.insert(0, PREFER_IPV4_STACK_IPV6_ADDRESS_EXCEPTION)
}
```